redis/server/install.sls

```diff
@@ -0,0 +1,104 @@
+include:
+  - redis.common
+
+{% from "redis/map.jinja" import redis_settings as r with context %}
+
+    {%- if r.install_from == 'source' %}
+
+redis_user_group:
+  group.present:
+    - name: {{ r.group }}g
+  user.present:
+    - name: {{ r.user }}g
+    - gid_from_name: True
+    - home: {{ r.home }}
+    - require:
+      - group: redis_group
+  file.managed:
+    - name: /etc/init/redis-server.conf
+    - template: jinja
+    - source: salt://redis/files/upstart.conf.jinja
+    - mode: '0750'
+    - user: root
+    - group: root
+    - context:
+        conf: /etc/redis/redis.conf
+        user: {{ r.user }}g
+        bin: {{ r.bin }}
+    - require:
+      - sls: redis.common
+
+redis-log-dir:
+  file.directory:
+    - name: /var/log/redis
+    - mode: 755
+    - user: {{ r.user }}g
+    - group: {{ r.group }}g
+    - makedirs: True
+    - require:
+      - user: redis_user_group
+
+        {%- if grains['os_family'] == 'Arch' %}
+redis-log-dir:
+  file.directory:
+    - name: /var/log/redis
+    - mode: 755
+    - user: {{ r.user }}
+    - group: {{ r.group }}
+    - makedirs: True
+        {%- endif %}
+
+    {%- endif %}
+
+redis_config:
+  file.managed:
+    - name: {{ r.cfg_name }}
+    - template: jinja
+    {% if r.source_path is not defined %}
+    - source: salt://redis/files/redis-{{ r.cfg_version }}.conf.jinja
+    {% else %}
+    - source: {{ r.source_path }}
+    {%- endif %}
+
+    {% if r.install_from == 'source' %}
+redis-initd:
+  file.managed:
+    - name: /etc/init.d/redis
+    - template: jinja
+    - source: salt://redis/files/redis_initd.jinja
+    - mode: '0777'
+    - user: root
+    - group: root
+    - require:
+      - file: redis_config
+    - require_in:
+      - file: redis_service
+    {%- endif %}
+
+    {% if r.disable_transparent_huge_pages is defined and r.disable_transparent_huge_pages %}
+redis_disable_transparent_huge_pages:
+    cmd.run:
+        - name: echo "never" > /sys/kernel/mm/transparent_hugepage/enabled
+
+    {%- endif %}
+redis_service:
+  service.{{ r.svc_state }}:
+    {% if r.install_from == 'source' %}
+    - name: {{ r.svc_name }}_{{ r.port }}
+    {% else %}
+    - name: {{ r.svc_name }}
+    {% endif %}
```

Comment:
Again, it may be [better for readability as a variable set outside the state](https://docs.saltstack.com/en/latest/topics/development/conventions/formulas.html#light-conditionals-and-looping)

Reply:
I am conflicted in this specific case.  To simplify lines Y and Z for readability I must add line X which balanced against each other is probably the same level of complexity as not setting a variable outside the state.  If there were three or more variables I would definitely use a variable.  But with two variables adding a next jinja statement to satisfy one rule is just moving complexity somewhere else (extra line to manage that variable).  Its subjective.